### PR TITLE
Lift + shift for cross-db macros

### PR DESCRIPTION
**Pull request by jtcohen6**
Follow-up to:
- https://github.com/dbt-labs/dbt-core/pull/5265
- https://github.com/dbt-labs/dbt-utils/pull/594
- https://github.com/dbt-labs/dbt-spark/pull/359

### CODE REVIEW

**Comment by jtcohen6** [comment]
Functional tests are running, which is actually amazing. The other tests are running into https://github.com/dbt-labs/dbt-core/issues/5268

**Comment by jtcohen6** [comment]
The idea would be:
- A new minor release, requiring dbt-core v1.2.0, with backwards compatibility for folks calling these macros directly — `spark_utils.dateadd`. I don't think that's many people, but still!
- A subsequent minor release that removes all backwards compatibility

File: dev-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pytest
+pyodbc==4.0.32
+git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-core&subdirectory=core
+git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-tests-adapter&subdirectory=tests/adapter
+git+https://github.com/dbt-labs/dbt-spark.git#egg=dbt-spark[ODBC]
```
Comment (dbeatty10) [inline dev-requirements.txt:5]: This is helpful!

File: pytest.ini [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[pytest]
+filterwarnings =
+    ignore:.*'soft_unicode' has been renamed to 'soft_str'*:DeprecationWarning
+    ignore:unclosed file .*:ResourceWarning
+env_files =
+    test.env
+testpaths =
+    tests/functional
```
Comment (dbeatty10) [inline pytest.ini:8]: Last Week I Learned (LWIL):
- [By default `pytest-dotenv` will not override any variables already defined in the process' environment](https://github.com/quiqua/pytest-dotenv/blob/866c90d9475bc469a0e9fc848868af0139658786/README.md#overriding-existing-values)

### Example
Let's say I have an environment variable defined locally:
`DBT_DATABASE_HOST=localhost`

And `test.env` contains:
`DBT_DATABASE_HOST=somethingelse`

By default, pytest-dotenv will keep the former and ignore the latter. Scroll down to see how to configure the opposite behavior.

### Override default behavior
So it depending on how we want pytest + pytest-dotenv to behave, we either will want to include the following (or not!):
```
env_override_existing_values = 1
```

**Review by dbeatty10 — APPROVED** [review]
You are amazing.

Love the creation of the `macros/dbt_utils/cross_db_utils/deprecated` folder. Separates things nicely.

File: pytest.ini (hunk 2 above)
Comment (jtcohen6) [reply pytest.ini:8]: A great LWIL!

This makes sense to me, insofar as I might have a `test.env`, but then also want to do:
```
export DBT_DATABASE_HOST=localhost
python3 -m pytest ...
```
I would indeed expect the env var I've just supplied to override the one in `test.env`

**Pull request merged**